airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from copy import deepcopy
+from typing import Any, Dict, Optional
+
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.types import Config, StreamSlice, StreamState
+
+
+class NewtoLegacyFieldTransformation(RecordTransformation):
+    """
+    Implements a custom transformation which maps a "new" field to the legacy equivalent of the field for streams which contain Deals and Contacts entities.
+    """
+
+    def __init__(self, field_mapping: Dict[str, str]) -> None:
+        self._field_mapping = field_mapping
+
+    def transform(
+        self,
+        record: Dict[str, Any],
+        config: Optional[Config] = None,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> Dict[str, Any]:
```

Comment:
The return type differs from the parent https://github.com/airbytehq/airbyte-python-cdk/blob/39786d26926d8b041df5d7b6ae70aeee68b84e41/airbyte_cdk/sources/declarative/transformations/transformation.py#L25

This means that the transformation is do in place and that the caller of this function should not store the result in a variable like https://github.com/airbytehq/airbyte/pull/48472/files#diff-6c388abccab93dc7577dc93da2d16f691f60bb9c806ec20011457fc10070b07cR708

It also probably means that we don't need to do a deep copy

Reply:
Good catch -- updated accordingly.